plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -730,6 +764,16 @@ compileExpr e = traceCompilation 2 ("Compiling expr:" GHC.<+> GHC.ppr e) $ do
     GHC.Var n
       | isProbablyBytestringEq n ->
           throwPlain $ UnsupportedError "Use of Haskell ByteString equality, possibly via the Haskell Eq typeclass"
+    GHC.Var n
+      -- Try to produce a sensible error message if a range like [1..9] is encountered.  This works
+      -- by looking for occurrences of GHC.Enum.enumFromTo and similar functions; the same error
+      -- occurs if these functions are used explicitly.
+      | isProbablyBoundedRange n ->
+          throwPlain $ UnsupportedError "Literal ranges are not supported: please use PlutusTx.Enum.enumFromTo or PlutusTx.Enum.enumFromThenTo"
```

Comment:
We don't know for sure that the user used list range syntax. You can instead say something like: "Unsupported feature: GHC.Enum.Enum. You probably used list range syntax, which is not currently supported. You can use PlutusTx.Enum.Enum instead".

Reply:
> We don't know for sure that the user used list range syntax.

Yes, I did worry about that but decided to go for the shorter version to avoid a giant string in the source code. I'll change it.